### ISSUE TITLE
Allow all child process' to finish before terminating parent

### ISSUE THREAD

**Issue by Dahaden** [boltpkg/bolt#195]
| Title               | Description
| --------------------|------------
| Version             | 0.22.1
| Type                | Issue
| node                | v10.9.0
| Operating System    | OSX
| Short Description   | Child processes can become orphaned when another child process exits with non-zero
| Detailed description| The first child to exit with non-zero, causes the main bolt process to exit with the same code. In cases where `--parallel` is used, this can leave many orphaned process which keep logging to terminal. In rare cases, orphaned processes can become zombies.

**Comment by Dahaden** [comment]
Perhaps failing on the first breaking test is something that would be wanted in certain situations. Would adding something like `--fail-fast=false` be a good new API to get around this change of behaviour?

**Comment by jamiebuilds** [comment]
Yeah I've wanted this for awhile. I think `--bail` and `--no-bail` would be the best flags (with --bail being default). The meow CLI we use supports this `--no-` syntax automatically

**Comment by jamiebuilds** [comment]
This should help you:

https://github.com/boltpkg/bolt/blob/d22fd7b44dcb79b4873e5899524f1231017e8736/src/utils/options.js#L35-L48
